### PR TITLE
developers onboarding action added

### LINKED ISSUE #538: Implement messages to celebrate successful PR contributions
Provide congratulatory messages when a new contributor submits their first and second PRs or more.

The Solution:-
We will add a GitHub Action that will trigger when a PR gets merged into the main branch.

I would love to implement it.

### PR DESCRIPTION
**Pull request by Mayank77maruti**
### PR Description

Closes #538 

This PR introduces a new GitHub Action that automatically posts a comment on a pull request once it has been merged into the main branch.

### Some notes

- [x] Please make sure that you placed the files in an appropriate folder
- [x] And that the files have indicative names.

- [x] Please note that Classiq runs automatic code linting, which may minorly alter some files.
- [] If you're familiar with `pre-commit`, you may run `pre-commit install`, and then at each commit, your files will be altered in a similar way

### CODE REVIEW

**Comment by amir-naveh** [comment]
Great, thanks! @TaliCohn - could you please review the text that is sent? I think we may want something more concise and related to the repo. Let me know if you want to discuss. Once you are good with it you can approve.

File: .github/workflows/developers_onboarding_notification.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+              }
+
+              function getMessage(count) {
+                const emoji = getRandomEmoji();
+                switch(count) {
+                  case 1:
+                    return `${emoji} **Fantastic work @${author}!** Your very first PR to ${repo} has been merged! 🎉🥳\n\n` +
+                           `You've just taken your first step into open-source, and we couldn't be happier to have you onboard. 🙌\n` +
+                           `If you're feeling adventurous, why not dive into another issue and keep contributing? The community would love to see more from you! 🚀\n\n` +
+                           `For any support, feel free to reach out to the developer onboarding lead: @${mention}. Happy coding! 👩‍💻👨‍💻`;
```
Comment (TaliCohn) [inline .github/workflows/developers_onboarding_notification.yml:71]: `${emoji} **Fantastic work @${author}!** Your very first PR to ${repo} has been merged! 🎉🥳\n\n` +
                           `If you're feeling adventurous, why not dive into another issue and keep contributing? The Classiq community would love to see more from you! 🚀\n\n` +
                     Happy coding! 👩‍💻👨‍💻`;

File: .github/workflows/developers_onboarding_notification.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                  case 2:
+                    return `${emoji} **Well done @${author}!** Two PRs merged already! 🎉🥳\n\n` +
+                           `With your second PR, you're on a roll, and your contributions are already making a difference. 🌟\n` +
+                           `Looking forward to seeing even more contributions from you. The developer onboarding lead: @${mention} is here if you need any help! Keep up the great work! 🚀`;
```
Comment (TaliCohn) [inline .github/workflows/developers_onboarding_notification.yml:75]: Great, please erase the "The developer onboarding lead: @${mention} is here if you need any help!"

File: .github/workflows/developers_onboarding_notification.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                  case 3:
+                    return `${emoji} **You're on fire, @${author}!** Three PRs merged and counting! 🔥🎉\n\n` +
+                           `Your consistent contributions are truly impressive. You're becoming a valued member of our community! 💖\n` +
+                           `Have you considered taking on some more challenging issues? We'd love to see what you can do! 💪\n\n` +
+                           `Remember, @${mention} is always here to support you. Keep blazing that trail! 🚀`;
```
Comment (TaliCohn) [inline .github/workflows/developers_onboarding_notification.yml:80]: erase: "Remember, @${mention} is always here to support you."

File: .github/workflows/developers_onboarding_notification.yml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                  case 5:
+                    return `${emoji} **High five, @${author}!** You've hit the incredible milestone of 5 merged PRs! 🖐️✨\n\n` +
+                           `Your dedication to ${repo} is outstanding. You're not just contributing code; you're shaping the future of this project! 🌠\n` +
+                           `We'd love to hear your thoughts on the project. Any ideas for new features or improvements? 🤔\n\n` +
+                           `@${mention} and the whole team applaud your efforts. You're a superstar! 🌟`;
```
Comment (TaliCohn) [inline .github/workflows/developers_onboarding_notification.yml:85]: `${emoji} **High five, @${author}!** You've hit the incredible milestone of 5 merged PRs! 🖐️✨\n\n` +
                           `Your dedication to ${repo} is outstanding. You're not just contributing code; you're shaping the future of quantum computing! 🌠\n` +
                           `We'd love to hear your thoughts on the project. Any ideas for new features or improvements? 🤔\n\n` +
                     You're a superstar! 🌟`;

File: .github/workflows/developers_onboarding_notification.yml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                  case 10:
+                    return `${emoji} **Double digits, @${author}!** 10 merged PRs is a massive achievement! 🏆🎊\n\n` +
+                           `Your impact on ${repo} is undeniable. You've become a pillar of our community! 🏛️\n` +
+                           `We'd be thrilled to have you take on a mentorship role for newer contributors. Interested? 🧑‍🏫\n\n` +
+                           `@${mention} and everyone here are in awe of your contributions. You're an open source hero! 🦸‍♀️🦸‍♂️`;
```
Comment (TaliCohn) [inline .github/workflows/developers_onboarding_notification.yml:90]: `${emoji} **Double digits, @${author}!** 10 merged PRs is a massive achievement! 🏆🎊\n\n` +
                           `Your impact on ${repo} is undeniable. You've become a pillar of our community! 🏛️\n` +
                   You're a quantum hero! 🦸‍♀️🦸‍♂️`;

File: .github/workflows/developers_onboarding_notification.yml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+                  default:
+                    if (count > 10) {
+                      return `${emoji} **Incredible, @${author}!** You've merged your ${count}th PR! 🎯🎊\n\n` +
+                             `Your ongoing commitment to ${repo} is truly remarkable. You're a driving force in our community! 🚀\n` +
+                             `Your contributions are helping to shape the future of this project. What exciting features or improvements do you envision next? 🔮\n\n` +
+                             `@${mention} and the entire team are grateful for your dedication. You're an inspiration to us all! 💫`;
```
Comment (TaliCohn) [inline .github/workflows/developers_onboarding_notification.yml:96]: `${emoji} **Incredible, @${author}!** You've merged your ${count}th PR! 🎯🎊\n\n` +
                             `Your ongoing commitment to ${repo} is truly remarkable. You're a driving force in our community! 🚀\n` +
                             `Your contributions are helping to shape the future of quantum computing! What exciting features or improvements do you envision next? 🔮\n\n` +
                             `We are grateful for your dedication! 💫`;

File: .github/workflows/developers_onboarding_notification.yml [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+                    } else {
+                      return `${emoji} **Great job, @${author}!** You've merged your ${count}th PR! 🎊\n\n` +
+                             `Your contributions to ${repo} are making a real difference. Keep up the fantastic work! 💪\n` +
+                             `Remember, every PR counts and helps improve the project. What will you tackle next? 🤔\n\n` +
+                             `@${mention} is here if you need any guidance. Onward and upward! 🚀`;
```
Comment (TaliCohn) [inline .github/workflows/developers_onboarding_notification.yml:101]: Erase: "`@${mention} is here if you need any guidance. Onward and upward! 🚀`;"

**Review by TaliCohn — CHANGES_REQUESTED** [review]

**Comment by Mayank77maruti** [comment]
Hey @TaliCohn , the pr is ready to be reviewed.
Please do review it.

**Comment by Mayank77maruti** [comment]
Hey @amir-naveh  , could you please review the pr.
I am participating in a opensource competition and it would be really great if you could review this pr.

**Review by TaliCohn — APPROVED** [review]

**Pull request merged**